src/import/helm.ts

```diff
@@ -0,0 +1,179 @@
+
+const MAX_HELM_BUFFER = 10 * 1024 * 1024;
+const CHART_SCHEMA = 'values.schema.json';
+const CHART_YAML = 'Chart.yaml';
+
+export class ImportHelm extends ImportBase {
+  public static async fromSpec(importSpec: ImportSpec): Promise<ImportHelm> {
+    const { source } = importSpec;
+    return new ImportHelm(source);
+  }
+
+  public readonly chartName: string;
+  public readonly chartUrl: string;
+  public readonly chartVersion: string;
+  public readonly chartSchemaPath: string | undefined;
+  public readonly chartDependencies: string[] = [];
+  public readonly tmpDir: string;
+
+  private constructor(source: string) {
+    super();
+
+    const [chartUrl, chartName, chartVersion] = getHelmChartDetails(source);
+
+    this.chartName = chartName;
+    this.chartUrl = chartUrl;
+    this.chartVersion = chartVersion;
+    this.tmpDir = pullHelmRepo(chartUrl, chartName, chartVersion);
+
+    const chartYamlFilePath = path.join(this.tmpDir, this.chartName, CHART_YAML);
+    const contents = Yaml.load(chartYamlFilePath);
+
+    if (contents && contents.length === 1 && contents[0].dependencies) {
+      for (const dependency of contents[0].dependencies) {
+        this.chartDependencies.push(dependency.name);
+      }
+    }
+
+    const potentialSchemaPath = path.join(this.tmpDir, this.chartName, CHART_SCHEMA);
+
+    if (fs.existsSync(potentialSchemaPath)) {
+      this.chartSchemaPath = potentialSchemaPath;
+    } else {
+      this.chartSchemaPath = undefined;
+    }
+  }
+
+  public get moduleNames() {
+    return [this.chartName];
+  }
+
+  protected async generateTypeScript(code: CodeMaker) {
+
+    emitHelmHeader(code);
+
+    let schema: JSONSchema4 | undefined;
+    if (this.chartSchemaPath !== undefined) {
+      schema = JSON.parse(fs.readFileSync(this.chartSchemaPath, 'utf-8'));
+    } else {
+      schema = undefined;
+    }
+
+    const types = new TypeGenerator({
+      definitions: schema?.definitions,
+    });
+
+    generateHelmConstruct(types, {
+      schema: schema,
+      chartName: this.chartName,
+      chartUrl: this.chartUrl,
+      chartVersion: this.chartVersion,
+      chartDependencies: this.chartDependencies,
+      fqn: this.chartName,
+    });
+
+    code.line(types.render());
+
+    cleanup(this.tmpDir);
+  }
+}
+
+/**
+ * Validating if a helm chart url is in an expected format
+ * @param url
+ */
+function validateHelmUrl(url: string): RegExpExecArray {
+  const helmRegex = /^helm:([A-Za-z0-9_.-:\-]+)\/([A-Za-z0-9_.-:\-]+)\@([0-9]+)\.([0-9]+)\.([A-Za-z0-9-+]+)$/;
+  const match = helmRegex.exec(url);
+
+  if (match) {
+    return match;
+  } else {
+    throw Error(`There was an error processing the helm chart url you passed in: ${url}. Make sure it matches the format of 'helm:<repo-url>/<chart-name>@<chart-version>'.`);
+  }
+}
+
+/**
+ * Gets information about the helm chart from the helm url
+ * @param url
+ * @returns chartUrl, chartName and chartVersion
+ */
+function getHelmChartDetails(url: string) {
```

Comment:
Make sure to use the right semantics for function names, it makes it easier to understand. (`get` is for getters, `fetch` is for http calls, `extract` is for regex and such...)

```suggestion
function extractHelmChartDetails(url: string) {
```